### PR TITLE
NEW: Run JS/CSS linting in Travis.

### PR DESCRIPTION
**Pull request by sminnee**
The ‘npm run lint’ command will be used to run listing on Travis, which
can also be used on local dev environments. These can also be used with
editor plugins to highlight errors immediately.

The intention is that this can be used in place of codeclimate. The
benefits are that we use a single toolchain in both CI and local dev,
which is not entirely the case at the moment.

Note the sass-lint is provided by “sudo gem install scss_lint”. It’s
possible that we can move to a node-based sass-lint; I can’t recall
what the motivation for using the scss_lint gem was - I think it was
mainly that we had the AirBNB styleguide already implemented as a linter
config.

----

### Merge blockers

 * [x] Move to the node-based `sass-lint` tool.

### CODE REVIEW

**Comment by chillu** [comment]
Looks like the `lwip` (sprity dep) compilation is throwing errors - so we should merge https://github.com/silverstripe/silverstripe-framework/pull/5918 first which removes that dep altogether.

Yeah, AirBnB's default to scsslint was one motivation, CodeClimate's support for scsslint only (no sass-lint) was another - good reason to remove CodeClimate as well :) It's not a merge blocker for this PR. There's a converter at http://sasstools.github.io/make-sass-lint-config/ - I've made a start:

```yml
# sass-lint config generated by make-sass-lint-config v0.1.2
#
# The following scss-lint Linters are not yet supported by sass-lint:
# PrivateNamingConvention

files:
  include: '**/client/src/**/*.scss'
  ignore:
    - 'client/src/styles/legacy/*'
    - 'admin/client/src/styles/legacy/*'
options:
  formatter: stylish
  merge-default-rules: false
rules:
  bem-depth: 1
  border-zero:
    - 1
    - convention: zero
  brace-style:
    - 1
    - allow-single-line: false
  class-name-format:
    - 1
    - convention: ^(?!js-).*
      convention-explanation: should not be written in the form js-*
  extends-before-declarations: 0
  extends-before-mixins: 0
  function-name-format: 1
  id-name-format:
    - 1
    - convention: hyphenatedbem
  leading-zero: 0
  mixin-name-format: 1
  mixins-before-declarations: 0
  no-extends: 1
  no-qualifying-elements: 0
  placeholder-name-format:
    - 1
    - convention: hyphenatedbem
  property-sort-order: 0
  quotes:
    - 1
    - style: double
  url-quotes: 0
  variable-name-format: 1
```

**Comment by sminnee** [comment]
Rebased, let's see what travis says.

**Comment by sminnee** [comment]
Ye gads, that's annoying.

scss-lint requires ruby 2, which isn't provided on Travis' PHP builders.

I'll fast-track the migration to the node-based tool and include it in this PR.

**Comment by sminnee** [comment]
OK this is all done now; npm sass-lint is used in place of the ruby-based scss-lint. It should be ready for merge!

File: .travis.yml [hunk 1]
```diff
@@ -48,8 +48,8 @@ before_script:
  - "if [ \"$DB\" = \"PGSQL\" ]; then composer require silverstripe/postgresql:2.0.x-dev --prefer-dist; fi"
  - "if [ \"$DB\" = \"SQLITE\" ]; then composer require silverstripe/sqlite3:2.0.x-dev --prefer-dist; fi"
  - "if [ \"$CMS_TEST\" = \"1\" ]; then COMPOSER_ROOT_VERSION=4.0.x-dev composer require silverstripe/cms:4.0.x-dev silverstripe/siteconfig:4.0.x-dev silverstripe/reports:4.0.x-dev --prefer-dist; fi"
+ - "if [ \"$NPM_TEST\" = \"1\" ] && [ `git log --format=format:%aE -1` = \"support@greenkeeper.io\" ]; then echo \"Removing npm-shrinkwrap.json\"; rm npm-shrinkwrap.json; fi"
```
Comment (chillu) [inline .travis.yml]: Heh, smart

**Comment by chillu** [comment]
Merged! I've also removed the CodeClimate webhook

**Comment by sminnee** [comment]
> Are you planning to create follow-up PRs for cms and asset-admin?

Yeah I can do that. The other thing that I should do is combine npm and behat, so that the impact of any package updates is tested via behat.

**Pull request merged**